Scaling-Service-Example/ScaleService.sh

```diff
@@ -0,0 +1,383 @@
+      else
+         if [[ "$CurrMin" -eq "$MaxMin" ]]; then
+            echo "Tried for $MaxMin minutes to start instance $1. Instance is taking too long to spin up"
+            echo "Look at ResourceStatus.txt for more information on the issue"
+            exit 1
+         fi
+         let CurrMin=CurrMin+1
+         sleep 60
+      fi
+   done
+}
+
+
+#################### Poll If Stopped Function ####################
+#
+#
+#
+# Parameters: $1 - The instance to poll
+##################################################################
+poll_if_stopped() {
+
+   CurrMin=1
+   MaxMin=15 #We will loop up to 15 mins to check if an instance has stopped
+
+   while [[ "$CurrMin" -le "$MaxMin" ]]; do
+      curl $CurlFlags -X GET --header "Accept: application/json" --header "authorization: $BearerToken" "$HOST"/organizations/"$ORGANIZATION"/spaces/"$SPACE"/serviceinstances/"$SERVICE_INSTANCE_ID"/resources/"$1" > ResourceStatus.txt
+      MachineStatus=$(tail -1 ResourceStatus.txt | jq .machinestatus | cut -f2 -d \")
+
+      if [[ "$MachineStatus" == "STOPPED" ]]; then
+         echo "Instance $1 stopped in $CurrMin minutes!"
+         rm ResourceStatus.txt
+         break
+      else
+         if [[ "$CurrMin" -eq "$MaxMin" ]]; then
+            echo "Tried for $MaxMin minutes to stop instance $1. Instance is taking too long to spin down"
+            echo "Look at ResourceStatus.txt for more information on the issue"
+            exit 1
+         fi
+         let CurrMin=CurrMin+1
+         sleep 60
+      fi
+   done
+}
+
+
+##################################################################
+# Main Function Start
+##################################################################
+
+# Before we execute the script, we need to make sure that jq utility is installed
+export JQ=$(command -v jq)
+if [[ -z "$JQ" ]]; then
+   echo "The jq utility has not been installed. Please read the script description and install jq."
+   exit 1
+fi
+
+echo "========================================================"
+echo "         Welcome to the WASaaS Scaling Script"
+echo "========================================================"
+echo ""
+echo "This script will perform the following actions:"
+echo ""
+echo "* List the nodes that are currently running "
+echo "* List the nodes that are currently stopped"
+echo "* Scale the service to the desired number of nodes"
+echo "* List nodes that have been started/stopped"
+echo ""
+echo --------------------------------------------------------
+echo "Begin ScaleService.sh - "$(date)
+echo --------------------------------------------------------
+
+# Import variables from properties file
+. ScaleService.properties
+echo "Host URL: $HOST"
+echo ""
+# Set other variables for use in script
+CurlFlags="-k -i -s"  # -k for ssl ignoring, -i for header info, -s for silent mode
+
+# Retrieve Bearer Token from Bluemix to use API
+echo "Retrieving bearer token..."
+echo ""
+get_bearer_token
+echo "Your bearer token is: $BearerToken"
```

Comment:
Yeah you're right. I copied this from Heather's script, but since she was the only one using her own script, it doesn't make sense for me to print out the user's Bearer token to the console.